AudioPlayer/AudioPlayer/player/AudioPlayerDelegate.swift

```diff
@@ -13,7 +13,7 @@ public typealias Metadata = [AVMetadataItem]
 
 /// This protocol contains helpful methods to alert you of specific events. If you want to be notified about those
 /// events, you will have to set a delegate to your `audioPlayer` instance.
-public protocol AudioPlayerDelegate: class {
+@objc public protocol AudioPlayerDelegate {
```

Comment:
funcs can be optional and obj-c compatible by defining them as `@objc optional func`
Should any of the current delegate funcs be optional?